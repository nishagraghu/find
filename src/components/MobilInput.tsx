import React, {useState, useRef} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  View,
  StatusBar,
  TouchableOpacity,
  
} from 'react-native';
import PhoneInput from 'react-native-phone-number-input';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import { useNavigation } from '@react-navigation/native';
import { Text } from 'react-native-paper';
import { RootState,useAppDispatch } from '../store/store';
import {sendOtp} from '../store/service/UserService';
const MobilInput : React.FC = () => {
  const [value, setValue] = useState('');
  const[errorFlag,setFlag] =useState(false);
  const [countryCode, setCountryCode] = useState('');
  const [formattedValue, setFormattedValue] = useState('');
  const [valid, setValid] = useState(false);
  const [disabled, setDisabled] = useState(false);
  const [showMessage, setShowMessage] = useState(false);
  const phoneInput = useRef<PhoneInput>(null);
  const navigation = useNavigation <any>();
  const dispatch   = useAppDispatch();
  const { formattedNumber  } = phoneInput.current?.getNumberAfterPossiblyEliminatingZero() || {};
  
  const submitPhone  = () =>{
     if(value.length!=10){
        
      setFlag(true)
      return;
     }
    
     dispatch(sendOtp('+91'+value));
    
     setFlag(false)

    // const checkValid = phoneInput.current?.isValidNumber(value);
    // setShowMessage(true);
    // setValid(checkValid ? checkValid : false);
    // setCountryCode(phoneInput.current?.getCountryCode() || '');
    
    navigation.navigate('Otp', { phoneNumber: '+91'+value } )
  }

  return (
    <View style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <View style={styles.container}>
        <SafeAreaView style={styles.wrapper}  >
          
          <PhoneInput 
            ref={phoneInput}
            defaultValue={value}
            defaultCode="IN"
            layout="first"
            onChangeText={(text) => {
              setValue(text);
            
            }}
            onChangeFormattedText={(text) => {
              
               setFormattedValue(text);
              
              //  setCountryCode(phoneInput.current?.getCountryCode() || '');
            }}
            countryPickerProps={{withAlphaFilter:true}}
            disabled={disabled}
            withDarkTheme
            withShadow
            autoFocus
          />
          {errorFlag &&<Text  testID="error-msg"  variant="labelLarge" style={styles.erorText} >please enter valid phone number</Text>}
          <TouchableOpacity
            style={styles.button}
            onPress={() => submitPhone()}>
            <Text   style={styles.buttonText}>SEND</Text>
          </TouchableOpacity>
          
        </SafeAreaView>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  erorText:{
    color:'red',
    marginVertical:10
    
  },
  container: {
    flex: 1,
    marginBottom: '10%',
    paddingBottom:'10%',
    backgroundColor: Colors.lighter,
  },
  wrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    marginTop: 20,
    height: 50,
    width: 300,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#1778F2',
    shadowColor: 'rgba(0,0,0,0.4)',
    shadowOffset: {
      width: 1,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 10,
  },
  buttonText:{
    color: 'white',
    fontSize: 14,
  },
  redColor: {
    backgroundColor: '#F57777'
  },
  message: {
    borderWidth: 1,
    borderRadius: 5,
    padding: 20,
    marginBottom: 20,
    justifyContent: 'center',
    alignItems: 'flex-start',
  },
});

export default MobilInput;