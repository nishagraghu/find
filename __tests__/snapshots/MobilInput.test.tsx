import React from 'react';
import { render } from '@testing-library/react-native';

import MobilInput from '../../src/components/MobilInput'



jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
      ...actualNav,
      useNavigation: jest.fn(),
    };
  });
describe('MobilInput component', () => {
  test('renders correctly', () => {
    const { toJSON } = render(<MobilInput />);
    expect(toJSON()).toMatchSnapshot();
  });
});
