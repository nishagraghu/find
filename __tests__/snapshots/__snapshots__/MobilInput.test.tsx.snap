// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobilInput component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
      "marginBottom": "10%",
      "paddingBottom": "10%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#F3F3F3",
        "flex": 1,
        "marginBottom": "10%",
        "paddingBottom": "10%",
      }
    }
  >
    <RCTSafeAreaView
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "white",
              "flexDirection": "row",
              "width": 600,
            },
            {
              "elevation": 10,
              "shadowColor": "rgba(0,0,0,0.4)",
              "shadowOffset": {
                "height": 5,
                "width": 1,
              },
              "shadowOpacity": 0.34,
              "shadowRadius": 6.27,
            },
            {},
          ]
        }
      >
        <View
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 32,
              "opacity": 1,
              "width": 150,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "marginRight": 10,
                "width": 30,
              }
            }
          >
            <ActivityIndicator
              size="small"
            />
          </View>
          <Modal
            animated={true}
            animationType="slide"
            hardwareAccelerated={false}
            onDismiss={[Function]}
            onRequestClose={[Function]}
            visible={false}
          />
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAi0lEQVRYR+3WuQ6AIBRE0eHL1T83FBqU5S1szdiY2NyTKcCAzU/Y3AcBXIALcIF0gRPAsehgugDEXnYQrUC88RIgfpuJ+MRrgFmILN4CjEYU4xJgFKIa1wB6Ec24FuBFiHELwIpQxa0ALUId9wAkhCnuBdQQ5ngP4I9wxXsBDyJ9m+8y/g9wAS7ABW4giBshQZji3AAAAABJRU5ErkJggg==",
              }
            }
            style={
              {
                "height": 14,
                "width": 12,
              }
            }
          />
        </View>
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#F8F9F9",
                "flex": 1,
                "flexDirection": "row",
                "paddingHorizontal": 30,
                "paddingVertical": 27,
                "textAlign": "left",
              },
              {},
            ]
          }
        >
          <TextInput
            autoFocus={true}
            editable={true}
            keyboardAppearance="dark"
            keyboardType="number-pad"
            onChangeText={[Function]}
            placeholder="Phone Number"
            selectionColor="black"
            style={
              [
                {
                  "color": "#000000",
                  "flex": 1,
                  "fontSize": 16,
                },
                {},
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#7CDB8A",
            "elevation": 10,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "rgba(0,0,0,0.4)",
            "shadowOffset": {
              "height": 5,
              "width": 1,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
            "width": 300,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              {
                "color": "white",
                "fontSize": 14,
              },
            ]
          }
        >
          Submit
        </Text>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
